### PR TITLE
chore: [#4] Integrated ESlint

### PR DESCRIPTION
**Pull request by eldarcelik**

### CODE REVIEW

**Comment by 4zuko** [comment]
What about using `.eslintignore`? Should we skip `node_modules` from the root?

**Comment by 4zuko** [comment]
Check whether you should add this to the configuration?

```json
settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
        moduleDirectory: ['node_modules', 'src'],
      },
    },
  },
```

**Comment by eldarcelik** [comment]
> Check whether you should add this to the configuration?
> 
> ```js
> settings: {
>   'import/resolver': {
>     node: {
>       extensions: ['.js', '.jsx', '.ts', '.tsx'],
>       moduleDirectory: ['node_modules', 'src'],
>     },
>   },
> },
> ```

I will add it to the eslintric.json file.
Explanation:
With this configuration, when you use import statements in the React project, ESLint will first look in the node_modules directory, and if it doesn't find the module there, it will look in the src directory. It will consider files with extensions .js, .jsx, .ts, and .tsx. It will try to resolve modules using the Node.js module resolution algorithm.

**Comment by eldarcelik** [comment]
> You can also use `import/order` rule that's exposed by `eslint-plugin-import` plugin (if it's not included through the recommended plugins).

I added some import/order rules in [commit](https://github.com/zuko4celik/movies-app-new/pull/20/commits/f79ab7b6c50f89421bc54ac2a799b2fd06d1b2a9)
Please check it and see if they are good and do I need to add some other rules.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
+        "eslint:recommended",
+        "plugin:react/recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:import/errors",
+        "plugin:import/warnings"
```
Comment (4zuko) [inline .eslintrc.json:11]: Per the documentation, it says:

```text
  - plugin:import/recommended
  # alternatively, 'recommended' is the combination of these two rule sets:
  - plugin:import/errors
  - plugin:import/warnings
````

That means that you can replace these two lines with `plugin:import/recommended`

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        "react",
+        "react-hooks",
+        "@typescript-eslint",
+        "import"
+    ],
+    "rules": {
+        "import/order": [
+            "error",
+            {
+                "newlines-between": "ignore",
```
Comment (4zuko) [inline .eslintrc.json:33]: It's recommended to use new line between import groups.

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                "groups": [
+                    "builtin",
+                    "external",
+                    "internal",
+                    ["parent", "sibling", "index"],
+                    "object"
+                ],
```
Comment (4zuko) [inline .eslintrc.json:40]: Below the groups add:

```js
pathGroups: [
  {
    pattern: 'react',
    group: 'external',
    position: 'before',
  },
],
pathGroupsExcludedImportTypes: ['react'],
```

That means the React imports will have priority over all **external** imports that are alphabetically sorted.

**Review by 4zuko — APPROVED** [review]
I'll approve the PR, but before you merge it to the **main** branch, please check the comments. Nice work. 💪

File: .eslintrc.json (hunk 2 above)
Comment (eldarcelik) [reply .eslintrc.json]: I added it inside [commit](https://github.com/zuko4celik/movies-app-new/pull/20/commits/111c24239002fb42784cca5cadad7697b599c6dd).

File: .eslintrc.json (hunk 3 above)
Comment (eldarcelik) [reply .eslintrc.json:39]: I added it in [commit 8](https://github.com/zuko4celik/movies-app-new/pull/20/commits/111c24239002fb42784cca5cadad7697b599c6dd).

**Pull request merged**